File: Python/299. Bulls and Cows.py
from collections import defaultdict
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        s_count = defaultdict(int)
        g_count = defaultdict(int)
        bulls = defaultdict(int)
        cows = 0
        for i in range(len(secret)):
            s_count[secret[i]] += 1
            g_count[guess[i]] += 1
            if secret[i] == guess[i]:
                bulls[secret[i]] += 1
        for k, v in s_count.items():
            cows += min(g_count[k], v) - bulls[k]
        return str(sum(bulls.values())) + "A" + str(cows) + "B"

File: 2021/299. Bulls and Cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        vector<int> countS(10, 0), countG(10, 0);
        int bulls = 0;
        for (int i = 0; i < secret.size(); i++) {
            if (secret[i] == guess[i]) {
                bulls++;
            } else {
                countS[secret[i] - '0']++;
                countG[guess[i] - '0']++;
            }
        }
        int cows = 0;
        for (int i = 0; i < 10; i++) {
            cows += min(countS[i], countG[i]);
        }
        return to_string(bulls) + "A" + to_string(cows) + "B";
    }
};